peekingduck/pipeline/nodes/input/visual.py

```diff
@@ -314,7 +282,40 @@ def _open_input(self, input_source: Any) -> None:
         else:
             self.videocap = VideoNoThread(input_source, self.mirror_image)
         self._fps = self.videocap.fps
-        self.total_frame_count = self.videocap.frame_count
+        self.total_frame_count = (
+            self.videocap.frame_count if self.videocap.frame_count > 0 else 0
+        )
```

Comment:
what about doing `max(0, self.videocap.frame_count)`?

Also, can `frame_count` be negative? Is that why this line got added?